### ISSUE TITLE
Allow the OSS admin user to regenerate their root token

### ISSUE THREAD

**Issue by russorat** [influxdata/influxdb#17329]
Since tokens cannot be changed, when we add new resources to the platform, the root token doesn't have access to it, even though the user has access to them. We need a way for an admin user to regenerate a new root token that has access to all the resources.

related:
https://github.com/influxdata/influxdb/issues/15961
https://github.com/influxdata/influxdb/issues/16120

**Comment by jademcgough** [comment]
@russorat An alternative to allowing the admin token to be regenerated would be to change the way that permissions are set in the admin token (instead of giving it access to each resource type individually, we just give it all permissions to all resource types forever). I think I'd prefer an approach like that, so that users don't need to manually regenerate their token when updating to a new version of InfluxDB, which is an awkward thing to ask them to do.

Would that also be acceptable?

**Comment by russorat** [comment]
@jademcgough makes sense to me. But what should users who already have root tokens created do?

**Comment by jsteenb2** [comment]
@russorat we need to explore this a bit more to see if the enforcement can be made retroactive. I am hopeful it can, but I'd have to take a closer look to confirm that.

I do think the title is misleading here. Rolling a new master token, feels like a suboptimal choice to fix the underlying issue.
